Added support for `header` keyword argument

fearphage opened the pull request:
Just converted another CLI param to a keyword argument.

dahlia reviewed (commented):
Could you write docs for this new option in *README.rst*? Thanks!

fearphage commented:
@dahlia Do you mind if I clean up the readme a bit? I was thinking of turning the options into a _sorted_ table to make finding entries more straight forward.

fearphage commented on README.rst:
```diff
@@ -59,111 +60,105 @@ The following is the full list of parameters.  Pass them as
+     -
+     - The iterable of extra raw options/arguments to pass to ``fzf``.
+
+       This is how you pass extra options that are not already defined
```
I added this sentence.

fearphage commented:
@dahlia Good to go. I added a note about #40 and added docs for the `header` param. Most of the rest is just reformatted/cleaned up copy and paste from the previous readme. 

One thing of note that I didn't find a quick fix for. The "API reference" link in the readme doesn't work in the `main` branch. I was going to fix it here, but I didn't quickly find out how to resolve that in the RST format. If you know what's wrong, I'd be happy to fix it while I'm here.

dahlia reviewed (approved):
Thanks for reformatting docs!
